client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -1442,6 +1442,15 @@ public TransactionSendResult sendMessageInTransaction(final Message msg,
         if (msg.getDelayTimeLevel() != 0) {
             MessageAccessor.clearProperty(msg, MessageConst.PROPERTY_DELAY_TIME_LEVEL);
         }
+        if (msg.getDelayTimeMs() != 0) {
+            MessageAccessor.clearProperty(msg, MessageConst.PROPERTY_TIMER_DELAY_MS);
```

Comment:
In this situation, it is recommended to throw an exception directly, so that the user can perceive the usage error.